xdsl/parser/attribute_parser.py

```diff
@@ -643,20 +645,98 @@ def _parse_optional_builtin_parametrized_attr(self) -> Attribute | None:
         self._consume_token(Token.Kind.BARE_IDENT)
         return parsers[name.text](name)
 
+    def _parse_builtin_dense_attr_hex(
+        self,
+        hex_string: str,
+        type: RankedVectorOrTensorOf[IntegerType]
+        | RankedVectorOrTensorOf[IndexType]
+        | RankedVectorOrTensorOf[AnyFloat],
+    ) -> tuple[list[int] | list[float], list[int]]:
+        """
+        Parse a hex string literal e.g. dense<"0x82F5AB00">, and returns its flattened data
+        and it's flattened shape, based on the parsed type.
+        Parse a hex string tensor literal, and returns its flatten data and its shape.
+
+        For instance, a dense<"0x82F5AB0182F5AB00"> attribute will return [28046722, 11269506]
+        for a tensor<2xi32> type.
+
+        only supports integer types that are multiple of 8, f32 and f64
+        """
+        element_type = type.element_type
+
+        # Strip off "0x" of hex string
+        stripped_string = hex_string[2:]
+
+        # Convert incoming hex to list of bytes
+        try:
+            byte_list = bytes.fromhex(stripped_string)
+        except ValueError:
+            self.raise_error("Hex string in denseAttr is invalid")
+
+        # Use struct builtin package for unpacking f32, f64
+        format_str: str = ""
+        match element_type:
+            case Float32Type():
+                chunk_size = 4
+                format_str = "@f"  # @ in format string implies native endianess
+            case Float64Type():
+                chunk_size = 8
+                format_str = "@d"
+            case IntegerType():
+                if element_type.width.data % 8 != 0:
+                    self.raise_error(
+                        "Hex strings for dense literals only support integer types that are a multiple of 8 bits"
+                    )
+                chunk_size = element_type.width.data // 8
+            case _:
+                self.raise_error(
+                    "Hex strings for dense literals are only supported for int, f32 and f64 types"
+                )
+        num_chunks = len(byte_list) // chunk_size
+
+        data_values: list[int] | list[float] = []
+
+        if isa(element_type, Float32Type | Float64Type):
+            for i in range(num_chunks):
+                parsed_float = struct.unpack(
```

Comment:
Should we use `unpack_from` or `iter_unpack`? feels like we could avoid the iteration in Python, and instead check that the number was what we expected

Reply:
Changed to using `unpack_from`, thanks for the suggestion!